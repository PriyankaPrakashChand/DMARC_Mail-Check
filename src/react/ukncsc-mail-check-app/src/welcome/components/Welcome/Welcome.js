import React, { Component } from 'react';
import { Divider, Form, Header } from 'semantic-ui-react';
import WelcomeSearchMessages from '../WelcomeSearchMessages';

export default class Welcome extends Component {
  componentDidMount = () => this.props.resetWelcomeSearch();

  componentDidUpdate = () => {
    const { searchResult, lastSearchTerm, history } = this.props;

    if (searchResult && searchResult.domainName === lastSearchTerm) {
      history.push(`/domain-security/${this.props.searchResult.id}`);
    }
  };

  addDomainOnClick = e => {
    e.preventDefault();
    this.props.fetchAddDomainToMailCheck(this.props.searchTerm);
  };

  searchOnChange = (e, data) =>
    this.props.updateWelcomeSearch({ searchTerm: data.value });

  showResultsOnSubmit = () =>
    this.props.fetchWelcomeSearch(this.props.searchTerm);

  render() {
    const {
      error,
      hasSearched,
      lastSearchTerm,
      loading,
      searchResult,
      searchTerm,
      searchTermIsPublicSectorOrg,
    } = this.props;

    return (
      <React.Fragment>
        <Header as="h1" style={{ marginTop: 10 }}>
          Mail Check
        </Header>
        <Divider hidden />
        <p style={{ maxWidth: 600 }}>
          Mail Check helps you understand how secure your email server
          configuration is, and how to improve and maintain it. The tool checks
          DMARC, SPF, and TLS configuration, and processes DMARC reports to help
          you understand them.
        </p>
        <Header as="h4" style={{ display: 'inline-block' }}>
          Check your email domain
        </Header>
        <Form onSubmit={this.showResultsOnSubmit}>
          <Form.Input
            fluid
            onChange={this.searchOnChange}
            style={{ maxWidth: 600 }}
            type="text"
            value={searchTerm}
          />
          <Form.Button color="blue" disabled={!searchTerm} loading={loading}>
            Show results
          </Form.Button>
        </Form>
        <Divider hidden />
        <WelcomeSearchMessages
          addDomainOnClick={this.addDomainOnClick}
          error={error}
          hasSearched={hasSearched}
          lastSearchTerm={lastSearchTerm}
          loading={loading}
          searchResult={searchResult}
          searchTermIsPublicSectorOrg={searchTermIsPublicSectorOrg}
        />
      </React.Fragment>
    );
  }
}
