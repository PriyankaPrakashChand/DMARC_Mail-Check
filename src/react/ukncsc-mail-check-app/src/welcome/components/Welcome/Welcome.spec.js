import React from 'react';
import { render } from 'react-testing-library';
import { MemoryRouter } from 'react-router-dom';
import Welcome from './Welcome';

describe('Welcome', () => {
  let container;

  describe('when displayed for the first time', () => {
    beforeEach(() => {
      ({ container } = render(
        <MemoryRouter>
          <Welcome
            lastSearchTerm=""
            resetWelcomeSearch={jest.fn()}
            searchResult={null}
            searchTerm=""
          />
        </MemoryRouter>
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());
  });

  describe('when the user has searched', () => {
    beforeEach(() => {
      ({ container } = render(
        <MemoryRouter>
          <Welcome
            hasSearched
            lastSearchTerm="foo.gov.uk"
            resetWelcomeSearch={jest.fn()}
            searchResult={{ id: 123, domainName: 'foo.gov.uk' }}
            searchTerm="foo.gov.uk"
            searchTermIsPublicSectorOrg
          />
        </MemoryRouter>
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());
  });
});
