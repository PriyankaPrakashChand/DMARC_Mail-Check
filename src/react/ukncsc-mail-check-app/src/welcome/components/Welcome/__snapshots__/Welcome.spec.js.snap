// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome when displayed for the first time it should match the snapshot 1`] = `
<div>
  <h1
    style="margin-top: 10px;"
    class="ui header"
  >
    Mail Check
  </h1>
  <div
    class="ui hidden divider"
  />
  <p
    style="max-width: 600px;"
  >
    Mail Check helps you understand how secure your email server configuration is, and how to improve and maintain it. The tool checks DMARC, SPF, and TLS configuration, and processes DMARC reports to help you understand them.
  </p>
  <h4
    style="display: inline-block;"
    class="ui header"
  >
    Check your email domain
  </h4>
  <form
    class="ui form"
  >
    <div
      class="field"
    >
      <div
        style="max-width: 600px;"
        class="ui fluid input"
      >
        <input
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="disabled field"
    >
      <button
        class="ui blue disabled button"
        disabled=""
        role="button"
        tabindex="-1"
      >
        Show results
      </button>
    </div>
  </form>
  <div
    class="ui hidden divider"
  />
</div>
`;

exports[`Welcome when the user has searched it should match the snapshot 1`] = `
<div>
  <h1
    style="margin-top: 10px;"
    class="ui header"
  >
    Mail Check
  </h1>
  <div
    class="ui hidden divider"
  />
  <p
    style="max-width: 600px;"
  >
    Mail Check helps you understand how secure your email server configuration is, and how to improve and maintain it. The tool checks DMARC, SPF, and TLS configuration, and processes DMARC reports to help you understand them.
  </p>
  <h4
    style="display: inline-block;"
    class="ui header"
  >
    Check your email domain
  </h4>
  <form
    class="ui form"
  >
    <div
      class="field"
    >
      <div
        style="max-width: 600px;"
        class="ui fluid input"
      >
        <input
          type="text"
          value="foo.gov.uk"
        />
      </div>
    </div>
    <div
      class="field"
    >
      <button
        class="ui blue button"
        role="button"
      >
        Show results
      </button>
    </div>
  </form>
  <div
    class="ui hidden divider"
  />
  <div
    style="margin: 0.2em 0em; max-width: 600px;"
    class="ui info message MailCheckMessage--container"
  >
    foo.gov.uk
    is not currently monitored by Mail Check.
    <a
      href=""
    >
      Add 
      foo.gov.uk
      to Mail Check
    </a>
    .
  </div>
  <div
    class="ui hidden divider"
  />
</div>
`;
